pkg/container/bitlpm/trie.go

```diff
@@ -185,6 +185,10 @@ func (t *trie[K, T]) Descendants(prefixLen uint, k Key[K], fn func(prefix uint,
 			currentNode.forEach(fn)
 			return
 		}
+		// currentNode is a leaf and has no children. Calling k.BitValueAt may overrun the key storage.
+		if currentNode.prefixLen >= t.maxPrefix {
```

Comment:
When did you run into this? In theory, this should be impossible given that `Upsert` sets the prefixLen of any key to at most the maxPrefix of the Trie.

Reply:
Ran into this with CIDR trie use. Got array out of bounds for an IPv4 CIDR when the bit 32 was asked for here.